### PR TITLE
DOC: fix PR02 errors in docstring for pandas.core.groupby.SeriesGroupBy.apply

### PR DESCRIPTION
**Pull request by jordan-d-murphy**
All PR02 Errors resolved in the following cases:

1. scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.core.groupby.SeriesGroupBy.apply

- [x] xref https://github.com/pandas-dev/pandas/issues/57111
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -214,12 +213,113 @@ def _get_data_to_aggregate(
...
+        a    2.0
+        b    1.0
+        dtype: float64
+
+        In the above, the groups are not part of the index. We can have them included
+        by using ``g2`` where ``group_keys=True``:
+
+        >>> g2.apply(lambda x: x * 2 if x.name == 'a' else x / 2)
+        a  a    0.0
+        a    2.0
```
Comment (mroeschke) [inline pandas/core/groupby/generic.py:300]: The alignment here look off?

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -1695,12 +1506,140 @@ def _aggregate_with_numba(self, func, *args, engine_kwargs=None, **kwargs):
...
+        2  1.000000  1.0
+
+        In the above, the groups are not part of the index. We can have them included
+        by using ``g2`` where ``group_keys=True``:
+
+        >>> g2[['B', 'C']].apply(lambda x: x / x.sum())
+                    B    C
+        A
+        a 0  0.333333  0.4
+        1  0.666667  0.6
```
[the remark this answers is not in the document]
Comment (jordan-d-murphy) [reply pandas/core/groupby/groupby.py:1605]: Great catch, thank you! I'm pushing a fix.

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (jordan-d-murphy) [reply pandas/core/groupby/generic.py:300]: Thanks - pushing a new commit to fix this.

**Comment by jordan-d-murphy** [comment]
@mroeschke can you confirm the alignment looks correct now?

**Review by mroeschke — COMMENTED** [review]
Thanks could you merge main once more? A new ruff docstring formatting was recently merged

**Comment by jordan-d-murphy** [comment]
Yup, I just did. Checks should be running again. Thanks for letting me know!

**Comment by jordan-d-murphy** [comment]
pre-commit.ci autofix

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Nice thanks for the quick turnaround @jordan-d-murphy

**Pull request merged**